datadog/metric_test.go

```diff
@@ -92,27 +86,31 @@ func makeTestServer(filename string) (*fixtureHandler, *httptest.Server) {
 }
 
 func TestStackdriverDataErrors(t *testing.T) {
+	ctx := context.Background()
+
+	storage := datastore.New(ctx)
+
 	handler, server := makeTestServer("")
 	defer server.Close()
 	m, _ := NewSourceMetric("metricname", &MetricConfig{Query: "metricquery"}, time.Second, time.Hour)
 	m.client.SetBaseUrl(server.URL)
 
 	// At this point HTTP server returns 404 to all requests, so we might as well test error handling.
-	_, _, err := m.StackdriverData(testCtx, time.Now().Add(-time.Minute), &datastore.StoredMetricRecord{})
+	_, _, err := m.StackdriverData(ctx, time.Now().Add(-time.Minute), &datastore.StoredMetricRecord{Storage: storage})
```

Comment:
You're not persisting any returned variables past this, so:

```go
if _, _, err := m.StackdriverData(...); err == nil {
  ...
}
```

Reply:
Done